src/app/modules/shared/services/template.service.ts

```diff
@@ -2,95 +2,134 @@ import { HttpParams } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { DataService } from '../data/data.service';
 import { Observable } from 'rxjs';
+import { environment } from '../../../../../src/environments/environment';
+
+interface EnvironmentConfig {
+  customAuth?: boolean;
+  baseUrl?: string;
+  qa?: string;
+  dev?: string;
+  bm?: string;
+  prod?: string;
+}
+
 @Injectable({
   providedIn: 'root'
 })
 export class TemplateService {
-  [x: string]: any;
-  templateFile:any;
-  templateError:any;
-  userSelectedFile:any;
-  constructor(private dataService: DataService) { }
+  private environmentUrls: EnvironmentConfig = {};
+  private currentEnvironment!: 'qa' | 'dev' | 'bm' | 'prod';
+
+  templateFile: any;
+  templateError: any;
+  userSelectedFile: any;
+
+  constructor(private dataService: DataService) {
+    // Load environment configuration from window.env and environment.ts
+    this.loadEnvironmentConfig();
+  }
+
+  // Method to load environment configuration from window.env and environment.ts
+  private loadEnvironmentConfig() {
+    this.environmentUrls = (window as any).env || {};
+
+    // Retrieve the current environment from environment.ts
+    const envKey = environment.currentEnvironment as 'qa' | 'dev' | 'bm' | 'prod';
```

Comment:
still in this case the code expects qa dev and bm env keys which should not be the case